### ISSUE TITLE
photos upload twice to cloudinary 

### ISSUE THREAD

**Issue by basantmohamed3** [cloudinary/cloudinary_npm#466]
I am trying to uplaod bulk of images to cloudinary but if upload two .. i find 4 photos on cloudinary I need to know whats wrong with my code 

//route
```
  app.post("/multiple-upload", multerMiddleware.array("files"), async (req, res) => {
  try {
    if (!req.files) {
      return res.status(200).json({ message: "Invalid file type" });
    }
    let uploadedFiles =  imageUploaderHandler({ files: req.files });
    return res.status(200).json({ message: "File stored", uploadedFiles });
  } catch (error) {
    console.log(error);
    return res
      .status(500)
      .json({ message: "Something went wrong while uplaoding this item" });
  }
})

const imageUploaderHandler = async ({ files }) => {
  let file = {};
  let uploadedFiles = [];
  let newImages = files.map(async (item) => {
   let result = await cloudinary.v2.uploader.upload(item.path)
      console.log(result);
      file.cloudinaryPublicId = result.public_id;
      file.fileUrl = result.secure_url;
      file.fileName = result.original_filename;
      uploadedFiles.push(file);
    
  });

  // return newImages;
  return await Promise.all(newImages);
};
```

**Comment by patrick-tolosa** [comment]
I see you've deleted our beautiful issue template, that's unfortunate as that information helps us help you much faster.

Your code seems to be right.
- Can you please confirm you're not sending a second request to your endpoint?

If this problem persists, can you please create a small git repo that reproduces this problem?

**Comment by basantmohamed3** [comment]
i confirm that I only send one request to the endpoint as the console only print the object one time

Kindly check my code in this link https://github.com/basantmohamed3/cloudinary-multipleUpload-issue

**Comment by momoip** [comment]
HI,
Your code has 2 app.post (line 70, 96) that each calls cloudinary.v2.uploader.upload.
This results in 2 upload actions and it shows from our log.
I hope this helps.
Best,
Mo

**Comment by patrick-tolosa** [comment]
Hey @basantmohamed3 

I ran your code and indeed the upload is happening twice.  (Thanks for providing the code, that was key to figuring this out)
After some debugging, I noticed you're `CloudinaryStorage`, which performs an upload internally

You're also explicitly uploading using the cloudinary node SDK (`..v2.upload(...)`).
You are uploading twice - once through CloudinaryStorage, and once manually.

Removing one of the methods should solve this issue for you.
